src/components/Editor/index.js

```diff
@@ -75,7 +87,8 @@ export default class Editor extends Component {
                     solution,                     
                     editorPosition: response.data.steps.length -1,
                     theActiveMathField: field,
-                    readOnly: this.props.match.params.action == 'view'
+                    readOnly: this.props.match.params.action == 'view',
+                    stepsFromLastSave: JSON.parse(JSON.stringify(response.data.steps))
```

Comment:
is JSON.parse(JSON.stringify(...)) needed here?

Reply:
yeah, to clone an array without any references